### DISCUSSION TITLE
question about uv__signal_lock

### DISCUSSION THREAD

**Discussion by theanarkh** [libuv/libuv#4444] [Q&A] [answered]
Hi All! I wonder why Libuv use `uv__signal_lock / uv__signal_unlock` instead of `uv_mutex_lock / uv_mutex_unlock` in. signal.c ?

**Comment by santigimeno** [comment, marked as answer]
I guess it's because we should be calling only async-signal-safe inside signal handlers and pthread_mutex_lock / unlock are not in the [list](https://man7.org/linux/man-pages/man7/signal-safety.7.html).

**Reply by theanarkh** [reply]
Oh! Thank you very much!
